applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAXNativeAdView.java

```diff
@@ -423,22 +424,29 @@ private void addMediaView(final MethodCall call)
 
         if ( mediaViewContainer == null )
         {
-            mediaViewContainer = new FrameLayout( context );
+            mediaViewContainer = new RelativeLayout( context );
             // Sets an identifier for the Google adapters to verify the view in the tree
             mediaViewContainer.setId( MEDIA_VIEW_CONTAINER_TAG );
             mediaViewContainer.setTag( MEDIA_VIEW_CONTAINER_TAG );
             nativeAdView.addView( mediaViewContainer );
         }
 
+        Rect rect = getRect( call );
+
         if ( mediaView.getParent() == null )
         {
-            mediaViewContainer.addView( mediaView );
-
-            mediaView.getLayoutParams().height = ViewGroup.LayoutParams.MATCH_PARENT;
-            mediaView.getLayoutParams().width = ViewGroup.LayoutParams.MATCH_PARENT;
+            RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(
+                    RelativeLayout.LayoutParams.MATCH_PARENT,
+                    RelativeLayout.LayoutParams.MATCH_PARENT );
+            mediaViewContainer.addView( mediaView, layoutParams );
+        }
+        else
+        {
+            mediaView.getLayoutParams().width = rect.width();
+            mediaView.getLayoutParams().height = rect.height();
```

Comment:
Are these LoC needed if now you're using RelativeLayout?

Reply:
You are absolutely right.  I somehow assumed that this is needed for the Google adapters.   Tested again with all the adapters.